query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotClientFactory.java

```diff
@@ -85,6 +85,10 @@ public ResultSetGroup executeQuery(String statement, Params params) {
       return preparedStatement.execute();
     }
 
+    public ResultSetGroup executeQuery(String statement) {
```

Comment:
Instead of adding this method and a new method `PinotBasedRequestHandler#resolveStatement`, why not modify the existing method `buildPreparedStatement` ?

Reply:
Wanted to keep that functionality once that gets fixed in pinot repo. But yes we can do that -- pass resolved statement and empty params to prepared statement -- but then the extra processing in PreparedStatement would be redundant.

Reply:
Does it mean the below `executeQueryAsync` method will continue to have the issue.

Reply:
fixed both the methods.